### PR TITLE
Play, earn, and fuel the Polkadot ecosystem with BlokkFaucet.

### PR DESCRIPTION
**Pull request by ooswald**
#### Project Abstract

This project develops browser-based games, enabling players to interact and earn rewards through a faucet system. A backend API manages game logic, faucet transactions, and user data. The reward pool is sustained by automated or manual transfers from a sponsor wallet, ensuring a continuous flow of incentives for players.

Leveraging the security and transparency of the Polkadot network, this project aims to create an engaging gaming experience that introduces users to blockchain technology. By combining gaming and blockchain, we seek to increase parachain activity and attract new users to the Polkadot ecosystem, exploring innovative applications of blockchain for gamification and user engagement.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (DOT, USDC & USDT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by ooswald** [comment]
I have read and hereby sign the Contributor License Agreement.

**Comment by ooswald** [comment]
recheck

File: applications/polkadot-faucet.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+|    0a. | License            | Apache 2.0                                                                                                               |
+|    0b. | Documentation      | The architecture and the designs will be documented.                                                                     |
+|    0c. | Testing Guide      | Core functionality will be covered by unit tests to ensure robustness.                                                   |
+|    0d. | Article            | We will publish an **article** that explains how the faucet works, why it was created, and how it can be used by anyone. |
+|     1. | Tech documentation | Architecture and implementation will be documented inline as well as in a documentation file.                            |
+|     2. | Usage documentaton | There will be a documentation how to use the faucet and game.                                                            |
+
+### Milestone 5 — Community Engagement
+
+**@stufi add the community engagement**
```
Comment (semuelle) [inline applications/polkadot-faucet.md:196]: It looks like there are deliverables missing in M5.

File: applications/polkadot-faucet.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+- **FTE:** 1,5
+- **Costs:** 2,400 USD
+
+| Number | Deliverable     | Specification                                                                                                             |
+| -----: | --------------- | ------------------------------------------------------------------------------------------------------------------------- |
+|    0a. | License         | Apache 2.0                                                                                                                |
+|    0b. | Documentation   | The designs and requirements will be documented.                                                                          |
+|    0c. | Testing Guide   | Core functionality will be covered by unit tests to ensure robustness.                                                    |
+|    0d. | Article         | We will publish an **article** that explains how the faucet works, why it was created, and how it can be used by anyone.  |
+|     1. | Design Document | A detailed description of the frontend design as well as an architecture diagram will be used to develop the blokkFaucet. |
```
Comment (semuelle) [inline applications/polkadot-faucet.md:141]: Please remove, the program does not fund the design phase of an app.

File: applications/polkadot-faucet.md [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+We are passionate about creating a fun and engaging gaming experience while also introducing users to the power of blockchain technology through Polkadot. We believe this project has the potential to attract new users to the Polkadot ecosystem and increase parachain activity. We are interested in exploring innovative ways to leverage blockchain technology for gamification and user engagement.
+
+### Project Details
+
+Polkadot faucet which provides a browser game secured by several OAuth providers and Cloudflare. The game management and faucet transactions will be handled by a secured backend API written in .NET. The game, transactions, and keys will be persisted in SQL Express databases. Transactions between are done via an external or selfhosted parachain node/API. The faucet wallet will be filled through manual or automated transactions from the sponsor wallet after several time and faucet wallet standings.
+
+#### Figma Mockups
+
+![Fruit Ninja](https://github.com/user-attachments/assets/e42507ca-094d-4bd0-9ab5-72e74013c89d)
+![Mining](https://github.com/user-attachments/assets/f0789d45-4186-4981-9a69-f6c377f526ca)
```
Comment (semuelle) [inline applications/polkadot-faucet.md:25]: Links are broken

File: applications/polkadot-faucet.md [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+- https://www.linkedin.com/in/r%C3%A9my-b%C3%B6hler-3565921a3/
+- https://www.linkedin.com/in/stephan-klaus-a3a90b50/
+
+## Development Roadmap
+
+### Overview
+
+- **Total Estimated Duration:** 2 months
+- **Full-Time Equivalent (FTE):** 1,5 FTE
+- **Total Costs:** 12,000 USD
```
Comment (semuelle) [inline applications/polkadot-faucet.md:127]: Please have a look at our updated application template. We offer a flexible (50%+) DOT payment, please provide your requested DOT portion (see [here](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md?plain=1#L127)).

File: applications/polkadot-faucet.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+
+- .NET
+- React
+- Tailwind
+- SQL Server Express
+- polkadotjs or API
+
+### Architecture Diagram
+
+![Architecture](https://github.com/blokk-studio/w3f-grants-program/assets/137090776/e295f4a5-b953-4141-9958-f9f34a5bbb33)
```
Comment (semuelle) [inline applications/polkadot-faucet.md:42]: Link broken as well

File: applications/polkadot-faucet.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+Polkadot faucet which provides a browser game secured by several OAuth providers and Cloudflare. The game management and faucet transactions will be handled by a secured backend API written in .NET. The game, transactions, and keys will be persisted in SQL Express databases. Transactions between are done via an external or selfhosted parachain node/API. The faucet wallet will be filled through manual or automated transactions from the sponsor wallet after several time and faucet wallet standings.
+
+#### Figma Mockups
+
+![Fruit Ninja](https://github.com/user-attachments/assets/e42507ca-094d-4bd0-9ab5-72e74013c89d)
+![Mining](https://github.com/user-attachments/assets/f0789d45-4186-4981-9a69-f6c377f526ca)
+
+See [all screen mocks in Figma](https://www.figma.com/design/6C57eHkcYOL3gmOhg9Skv0/POW-Faucet-Game?node-id=1%3A27&t=QqnRK1Y0J9lNYL6o-1).
+
+You can access a [clickable Figma prototype](https://www.figma.com/proto/6C57eHkcYOL3gmOhg9Skv0/POW-Faucet-Game?page-id=1%3A27&node-id=1-28&viewport=410%2C293%2C0.17&t=4JhvjAeBiVyBlpeo-1&scaling=min-zoom&starting-point-node-id=1%3A28).
```
Comment (semuelle) [inline applications/polkadot-faucet.md:29]: What does 'Mining' refer to in this UI?

**Review by semuelle — CHANGES_REQUESTED** [review]
Thank you for the application, @ooswald. Just FYI, the document was committed by a non-Github account, so you won't be able sign the CLA. But I take your comment from the account with the same name attached as approval. 

I left a few change requests below. But just to clarify, do you want to build a suite of games that rewards players with tokens, or is this a way to entertain users while they wait for their testnet faucet drop? For the former, would these tokens be provided by sponsors? The latter doesn't make much sense as there is basically no wait time on the main testnet faucets.

File: applications/polkadot-faucet.md (hunk 6 above)
Comment (stephan3000) [reply applications/polkadot-faucet.md:29]: It is not real mining, it is just a working title for this UI

**Comment by stephan3000** [comment]
> Thank you for the application, @ooswald. Just FYI, the document was committed by a non-Github account, so you won't be able sign the CLA. But I take your comment from the account with the same name attached as approval.
> 
> I left a few change requests below. But just to clarify, do you want to build a suite of games that rewards players with tokens, or is this a way to entertain users while they wait for their testnet faucet drop? For the former, would these tokens be provided by sponsors? The latter doesn't make much sense as there is basically no wait time on the main testnet faucets.

The primary goal of these games is to provide entertainment and a bit of fun while users wait for faucet drips. The idea is to prevent bots from dominating these faucets by incorporating an element of human interaction and engagement.  This way, we can ensure that real users have a fair chance to get tokens.

**Comment by keeganquigley** [comment]
Hi @stephan3000 thanks for your answers.

If the goal is to prevent bots from spamming the faucet, doesn't a simple CAPTCHA test already serve this purpose? As the official [Paseo Faucet](https://faucet.polkadot.io/) already offers one. 

Additionally, would this be just for valueless testnet tokens? The application suggests that you want to pool funds to give real DOT rewards.

I get that a game can be fun, but I agree that there is no wait time when using the faucet, unless you are referring to the wait time to request more tokens. In my experience helping people use the faucets, most users want to grab their tokens and go test something. I'm not sure that many would have the patience for a game, to win valueless tokens.

**Comment by stephan3000** [comment]
> Hi @stephan3000 thanks for your answers.
> 
> If the goal is to prevent bots from spamming the faucet, doesn't a simple CAPTCHA test already serve this purpose? As the official [Paseo Faucet](https://faucet.polkadot.io/) already offers one.
> 
> Additionally, would this be just for valueless testnet tokens? The application suggests that you want to pool funds to give real DOT rewards.
> 
> I get that a game can be fun, but I agree that there is no wait time when using the faucet, unless you are referring to the wait time to request more tokens. In my experience helping people use the faucets, most users want to grab their tokens and go test something. I'm not sure that many would have the patience for a game, to win valueless tokens.

You've raised some excellent points, and I appreciate the critical feedback.

You're absolutely right that CAPTCHA tests are a standard and effective way to deter bots. However, we believe that our approach offers a more engaging and user-friendly alternative. Our games aim to make the waiting process more enjoyable while still achieving the goal of bot prevention. Think of it as a more interactive and fun CAPTCHA.

The initial concept did explore the possibility of pooling funds for DOT rewards, but that is not a priority at this stage. The primary focus is on testnet tokens, which are indeed valueless in a monetary sense. You're correct that the wait times on the mainnet faucet are minimal. However, this project is primarily aimed at testnets with longer drip times or limited token availability. In those scenarios, we believe that offering a fun distraction in the form of games could be a welcome addition for users who might otherwise get bored or impatient while waiting.

We acknowledge that not all users will be interested in playing games, and that's perfectly fine. The games are intended to be an optional feature for those who want to pass the time in a more engaging way. The core functionality of the faucet would remain unaffected, allowing users to quickly claim their tokens and move on if they prefer.

We appreciate your insights and will definitely take them into consideration. Our goal is to strike a balance between bot prevention, user engagement, and overall user experience.

**Comment by keeganquigley** [comment]
Thanks for elaborating @stephan3000 the approach makes much more sense to me now. Glad to hear that it would also only be for testnets and be optional. I guess that would lead me to a couple more questions:

- Do you already have some projects in mind that could utilize this app?
- What are some examples of testnets that have a limited token availability?
- Would collaborate with each project to implement your app into their faucets, or is the idea to create a website where you can integrate faucets for all of these projects?

**Comment by stephan3000** [comment]
> * Do you already have some projects in mind that could utilize this app?
> * What are some examples of testnets that have a limited token availability?
> * Would collaborate with each project to implement your app into their faucets, or is the idea to create a website where you can integrate faucets for all of these projects in one place?

We don't have specific projects formally lined up yet, but we'll search collaborations with several parachain teams and testnet initiatives within the Polkadot ecosystem. Some promising candidates include:

- Westend: The primary testnet for Polkadot and Kusama, attracting a large number of users who need WND tokens for testing and experimentation.
- Rococo Testnet: This central testnet for parachains could also leverage our app to distribute tokens and incentivize user participation.
- Various Parachain Testnets: Many parachains like Acala, Moonbeam, and Astar have their own testnets with limited token supplies, making them ideal candidates for integration.
The above mentioned testchains often have restrictions on token distribution to simulate real-world scarcity and incentivize meaningful interactions. Our app aims to add value in these scenarios by making the process of acquiring testnet tokens more enjoyable and rewarding.

Our preferred approach is to create a centralized Multifaucet platform where various faucets could be accessed and managed. This could be a valuable resource for users who want to explore different testnets and parachains.

We're excited about the potential of this project to enhance the Polkadot ecosystem and create a more engaging and rewarding experience for users.

**Comment by keeganquigley** [comment]
Thanks @stephan3000 for your reply. Finally, if you were willing to lower the grant by $2k, it would become a level 1 grant and only need two approvals instead of three. But you are also welcome to leave as is.

**Comment by stephan3000** [comment]
@keeganquigley We appreciate the suggestion, but we're already quite tight on budget and feel that reducing the grant would compromise our ability to deliver the project's properly.

**Comment by keeganquigley** [comment]
Fair enough thanks @stephan3000 I will mark the application as ready for review so the rest of the committee members can comment.

**Review by semuelle — COMMENTED** [review]
Thanks for the updates, @ooswald.

> * The above mentioned testchains often have restrictions on token distribution to simulate real-world scarcity and incentivize meaningful interactions.

I justed tested the Rococo, Westend, Shibuya and Moonbase faucets, and none of the transactions took longer than nine seconds. I think there is merit in the idea to gamify faucets and to make the experience more enjoyable, but currently I don't see a need for this with the current experience.

**Comment by keeganquigley** [comment]
Thanks @stephan3000 I also have to agree that I don't see a need for this at this time. Therefore I'm not willing to approve it as is, but I wish you the best of luck in finding funding moving forward.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks a lot for the application and all the effort you put into it. However, the grants committee decided today not to go ahead with it, mostly because of the reasons shared with you above. Independent of it, we wish you all the best for your project. Let us know if you have any questions.

**Pull request closed without merging**